gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -883,3 +894,29 @@ message TranslatedString {
   // The following extension IDs are reserved for private use by any organization.
   extensions 9000 to 9999;
 }
+
+
+// Describes the physical path that a vehicle takes when it's not part of the (CSV) GTFS,
+// such as for a detour. Shapes belong to Trips, and consist of a sequence of shape points.
+// Tracing the points in order provides the path of the vehicle.  Shapes do not need to intercept
+// the location of Stops exactly, but all Stops on a trip should lie within a small distance of
+// the shape for that trip, i.e. close to straight line segments connecting the shape points
+// NOTE: This message is still experimental, and subject to change. It may be formally adopted in the future.
+message Shape {
+  // Identifier of the shape. Must be different than any shape_id defined in the (CSV) GTFS.
+  // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
+  required string shape_id = 1;
```

Comment:
I believe these two fields should be `optional` in the .proto file due to quirks in proto2 protocol buffers. See "Required Is Forever" at https://developers.google.com/protocol-buffers/docs/proto#specifying_field_rules. The reference docs can and should remain "Required".